synapse/appservice/__init__.py

```diff
@@ -400,6 +401,7 @@ def __init__(
         to_device_messages: List[JsonDict],
         one_time_key_counts: TransactionOneTimeKeyCounts,
         unused_fallback_keys: TransactionUnusedFallbackKeys,
+        device_list_summary: DeviceLists,
```

Comment:
(I chose a semi-random spot for this comment)

I'm surprised how much of the code needs to have the new `device_list_summary` threaded though it -- should we be creating an `AppServiceTransaction` (or some other object) high up in the stack and passing that along instead?

If so, could probably be done as a follow-up...especially if we're going to be adding more data to transactions.